src/tests/baseservices/exceptions/stackoverflow/stackoverflowtester.cs

```diff
@@ -40,19 +40,25 @@ static bool TestStackOverflow(string testName, string testArgs, out List<string>
 
             stderrLines = lines;
 
-            int expectedExitCode;
+            int[] expectedExitCodes;
             if ((Environment.OSVersion.Platform == PlatformID.Unix) || (Environment.OSVersion.Platform == PlatformID.MacOSX))
             {
-                expectedExitCode = 128 + 6;
+                expectedExitCodes = new int[] { 128 + 6};
             }
             else
             {
-                expectedExitCode = unchecked((int)0xC00000FD);
+                expectedExitCodes = new int[] { unchecked((int)0xC00000FD), unchecked((int)0x800703E9) };
             }
 
-            if (testProcess.ExitCode != expectedExitCode)
+            if (!Array.Exists(expectedExitCodes, code => testProcess.ExitCode == code))
             {
-                Console.WriteLine($"Exit code: 0x{testProcess.ExitCode:X8}, expected 0x{expectedExitCode:X8}");
+                string separator = string.Empty;
+                StringBuilder expectedListBuilder = new StringBuilder();
+                Array.ForEach(expectedExitCodes, code => {
```

Comment:
nit: the ForEach predicate generates 4-5x extra code, a simple for loop would be cleaner in this case.

Array.ForEach: https://godbolt.org/z/fqnv6Kxqd
for loop: https://godbolt.org/z/vETrWG45r

Reply:
It's a test so I assume we don't care about code size

Reply:
Yea just a nit; Array.ForEach is not common in test code either and simple for/while loops are a bit cleaner :)

Reply:
I prefer keeping it as is. It is just an error message generation code and it seems more readable than an explicit loop to me.